docs/docs/installation/build-from-source/index.md

```diff
@@ -42,38 +60,84 @@
 1. Press <kbd>Ctrl</kbd>+<kbd>D</kbd> to exit the `psql` shell.
 
 
-!!! note "Default Directory"
-    We will be using the home directory(accessed by `~/`) as the default working directory. 
+### Set up your installation directory
+
+1. Choose a directory to store the Mathesar application files.
+
+    !!! example "Examples"
+        - `/home/my_user_name/mathesar`
+        - `/etc/mathesar`
+
+1. Type your installation directory into the box below. Do not include a trailing slash.
+
+    <input data-input-for="MATHESAR_INSTALLATION_DIR" aria-label="Your Mathesar installation directory"/>
+
+    Then press <kbd>Enter</kbd> to customize this guide with your installation directory.
+
+1. Create your installation directory.
+
+    ```
+    mkdir -p xMATHESAR_INSTALLATION_DIRx
+    ```
+
+    !!! note "When installing outside your home folder"
+        If you choose a directory outside your home folder, then you'll need to create it with `sudo` and choose an appropriate owner for the directory (i.e. `root` or a custom user of your choosing).
+        
+        The remainder of this guide requires you to **run commands with full permissions inside your installation directory**. You can do this, for example via:
+
+        - `chown my_user_name: xMATHESAR_INSTALLATION_DIRx`
+
+            Or
+
+        - `sudo su`
+
+1. Navigate into your installation directory.
+
+    ```
+    cd xMATHESAR_INSTALLATION_DIRx
+    ```
+
+    The remaining commands in this guide should be run from within your installation directory.
 
-### Set up the Environment
 
-1. Clone the Mathesar repo.
+### Set up the environment
+
+1. Clone the git repo into a directory where you will install Mathesar, and `cd` into that directory.
 
     ```sh
-    cd ~/
-    git clone https://github.com/centerofci/mathesar.git
+    git clone https://github.com/centerofci/mathesar.git .
+    ```
+
+1. Switch to the `master` branch to install the latest stable release.
+
+    ```
+    git checkout master
     ```
 
+    !!! warning "Important"
+        If you don't run the above command you'll end up installing the latest _development_ version of Mathesar, which will be less stable.
+
+    !!! tip
+        You can install a specific Mathesar release by running commands like `git checkout 0.1.1` (to install version 0.1.1, for example). You can see all available versions by running `git tag`.
 
 1. We need to create a python virtual environment for the Mathesar application.
 
     ```sh
-    python3.9 -m venv ~/mathesar/mathesar-venv
+    python -m venv ./mathesar-venv
```

Comment:
Also the initial `./` is extraneous.

Reply:
I can understand the problem with the ambiguous python version.

- But I don’t understand the reason for suggesting `virtualenv`. `virtualenv` uses the [default python](https://virtualenv.pypa.io/en/latest/user_guide.html#python-discovery) when creating the virtual environment unless the python binary is specified using the `-p` flag. How is it any different from your alternate suggestion(creating the `venv using the correct python binary)

Reply:
In a separate channel, we've agreed to the compromise solution (i.e., specify the path and avoid the need for `virtualenv`.

Reply:
@mathemancer 

Even though I'm out this week I wanted to briefly chime in with a response to this...

> Also the initial `./` is extraneous.

I understand it's extraneous, but I added this because I think it's helpful for readers who don't understand these steps as much. The argument `./mathesar-venv` pretty clearly looks like a path, whereas the argument `mathesar-venv` looks like it could potentially be some other string with special meaning. In the same vein, I also changed `.` to `source` in another command. Sure it's a little longer, but it helps the reader to better understand what's going on. This is not to say that I think clarity always trumps brevity. It's a balance. But in these cases the extra characters are minimal enough that I think they add value.